crates/superposition/src/app_state.rs

```diff
@@ -52,6 +61,40 @@ pub async fn get(
         _ => Some(kms::new_client().await),
     };
 
+    cfg_if::cfg_if! {
+        if #[cfg(feature = "high-performance-mode")] {
+            let redis_url =
+                get_from_env_or_default("REDIS_URL", String::from("http://localhost:6379"));
+            let redis_pool_size = get_from_env_or_default("REDIS_POOL_SIZE", 10);
+            let redis_max_attempts = get_from_env_or_default("REDIS_MAX_ATTEMPTS", 10);
+            let config = fred::types::RedisConfig::from_url(&redis_url)
+                .expect(format!("Failed to create RedisConfig from url {}", redis_url).as_str());
+            let reconnect_policy =
+                fred::types::ReconnectPolicy::new_constant(redis_max_attempts, 100);
+            let redis_pool = fred::clients::RedisPool::new(
+                config,
+                Some(PerformanceConfig {
+                    auto_pipeline: true,
+                    ..Default::default()
+                }),
+                Some(ConnectionConfig {
+                    connection_timeout: Duration::from_millis(1000),
+                    ..Default::default()
+                }),
+                Some(reconnect_policy),
+                redis_pool_size,
+            )
+            .map_err(|e| format!("Could not connect to redis due to {e}"))
```

Comment:
Can we log it as well ?

Reply:
It would panic on the unwrap, and fail the boot up so it will be logged